src/hooks/useSendEmailVerification.ts

```diff
@@ -0,0 +1,35 @@
+import { Auth, AuthError, sendEmailVerification as fbSendEmailVerification } from 'firebase/auth'
+import { useCallback, useState } from 'react'
+
+export type SendEmailVerificationHook = [
+  () => Promise<boolean>,
+  boolean,
+  AuthError | Error | undefined,
+]
+
+const useSendEmailVerification = (auth: Auth): SendEmailVerificationHook => {
+  const [error, setError] = useState<AuthError>()
+  const [loading, setLoading] = useState<boolean>(false)
```

Comment:
different namings in diff places
some strings before: `const [isLoading, setIsLoading] = useState(true)` but without type of state)

Reply:
partially fixed in the last commit